synchronization/latch.cc

```diff
@@ -0,0 +1,32 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/synchronization/latch.h"
+
+namespace flutter {
+
+Latch::Latch(size_t count) : initial_(count), count_(count) {}
+
+Latch::~Latch() = default;
+
+void Latch::Wait() {
+  if (count_ > 0) {
+    ftl::MutexLocker lock(&condition_mutex_);
+    while (count_ != 0) {
```

Comment:
What would happen if multiple CountDown() calls decrease count_ below zero before Wait() can read count_?

Wait() should check for count_ > 0, or CountDown() should not decrement if count_ is already zero